File: M07_BounceGravity/app/src/main/java/com/example/m03_bounce/Box.java
package com.example.m03_bounce;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;

/**
 * Created by Russ on 08/04/2014.
 */
public class Box {

    int xMin, xMax, yMin, yMax;
    private Paint paint;  // paint style and color
    private Rect bounds;
    private Bitmap background;

    public Box(int color, Context context, int drawableResId) {
        paint = new Paint();
        paint.setColor(color);
        bounds = new Rect();
        background = BitmapFactory.decodeResource(context.getResources(), drawableResId);
    }

    public void set(int x, int y, int width, int height) {
        xMin = x;
        xMax = x + width - 1;
        yMin = y;
        yMax = y + height - 1;
        // The box's bounds do not change unless the view's size changes
        bounds.set(xMin, yMin, xMax, yMax);
    }

    public void draw(Canvas canvas) {
        canvas.drawBitmap(background, null, bounds, paint);
    }
}

File: M03_Bounce/app/src/main/java/com/example/m03_bounce/Box.kt
package com.example.m03_bounce

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect

/**
 * Created by Russ on 08/04/2014.
 */
class Box(color: Int) {
    var xMin: Int = 0
    var xMax: Int = 0
    var yMin: Int = 0
    var yMax: Int = 0
    private val paint: Paint // paint style and color
    private val bounds: Rect

    init {
        paint = Paint()
        paint.setColor(color)
        bounds = Rect()
    }

    fun set(x: Int, y: Int, width: Int, height: Int) {
        xMin = x
        xMax = x + width - 1
        yMin = y
        yMax = y + height - 1
        // The box's bounds do not change unless the view's size changes
        bounds.set(xMin, yMin, xMax, yMax)
    }

    fun draw(canvas: Canvas) {
        canvas.drawRect(bounds, paint)
    }
}
